### ISSUE TITLE
Rails 7 support

### ISSUE THREAD

**Issue by jasonfb** [alassek/activerecord-pg_enum#25]
[PGEnum] Current ActiveRecord version unsupported! Falling back to: 6.1

**Comment by alassek** [comment]
Native enums are supported by 7, so I'll probably do one more release to officially deprecate this library so people can migrate their schemas.

**Comment by soffes** [comment]
@alassek if you have the time, it would really appreciate a guide for how to adopt the Rails 7 things. I was able to figure out the model changes, but I'm having trouble finding how to update the migrations other using using plain SQL.

For others, this is what I in my models:

```ruby
KINDS = %w[foo bar]

# Before
include PGEnum(kind: KINDS, _suffix: true)

# After
enum :kind, KINDS.index_by(&:to_sym), suffix: true
```

**Comment by alassek** [comment]
@soffes thank you, I glad that it was useful! Lately I've been much happier working with [ROM](https://rom-rb.org/), so it's just as well that Rails is obsoleting this gem.

The major difference is that my syntax was `t.enum :column_name, as: "sql_type"` and rails is `t.enum :column_name, enum_type: "sql_type"`. I think mine reads better but whatever.

As for the module builder, my recommendation is to just [copy-paste the code into your project](https://github.com/alassek/activerecord-pg_enum/blob/main/lib/active_record/pg_enum.rb#L64-L92). It's a very small part of the gem, not really worth the dependency on its own.

**Comment by rayfaddis** [comment]
Rails 7 added better enum model support for PG enums though only added the `create_enum` migration command so this gem still offers\offered some nice and additional things.

See my Rails 7 PR that adds all of the migration commands to Rails 7 https://github.com/rails/rails/pull/44898

**Comment by alassek** [comment]
@rayfaddis That's a fair point. I will release a new version with only the additional stuff.

**Comment by alassek** [comment]
Version 2.0 has been released for Rails 7, there are a couple breaking interface changes. See CHANGELOG for more info

**Comment by jasonfb** [comment]
@alassek - can you explain what advantage this gem has over the native Rails 7 tools?

**Comment by alassek** [comment]
@jasonfb Everything that Rails 7 has implemented has been dropped from 2.0 but there were some additional migration helpers that they didn't implement:

- `drop_enum`
- `rename_enum`
- `add_enum_value`
- `rename_enum_value`

**Comment by andyweber-bamf** [comment]
Rails 7.1 adds the helpers:

- drop_enum
- rename_enum
- add_enum_value
- rename_enum_value
